crates/nu-command/src/filesystem/ucp.rs

```diff
@@ -0,0 +1,709 @@
+                            None,
+                            Vec::new(),
+                        ))
+                    }
+                }
+            }
+        }
+        let options = uu_cp::Options {
+            attributes_only,
+            backup: backup_mode,
+            copy_contents,
+            cli_dereference,
+            copy_mode,
+            // dereference,
+            dereference: !(no_dereference || no_dereference_preserve_links || archive || recursive)
+                || dereference,
+            no_target_dir,
+            one_file_system: false,
+            overwrite,
+            parents,
+            sparse_mode,
+            strip_trailing_slashes,
+            reflink_mode,
+            // attributes,
+            attributes,
+            recursive,
+            backup_suffix: suffix,
+            target_dir,
+            // target_dir: None,
+            update: update_mode,
+            debug,
+            verbose: verbose || debug,
+            progress_bar: progress,
+        };
+
+        // For enabling easy `crawl` of cp command, we need to strip current directory
+        // as it looks like uu_cp takes relative directory for sources & target,
+        // but nushell always resolves to the $CWD
+        // again, kind of a hack
+        let paths = paths
+            .iter()
+            .map(|path| match path.strip_prefix(&current_dir_path) {
+                Ok(p) => {
+                    if p.to_str()?.is_empty() {
+                        Some(PathBuf::from("."))
+                    } else {
+                        Some(p.to_path_buf())
+                    }
+                }
+                Err(_) => Some(path.to_path_buf()),
+            })
+            // I guess this is caught by nu, if not
+            // we handle it here.
+            .collect::<Option<Vec<PathBuf>>>()
+            .ok_or(ShellError::GenericError(
+                "Not valid UTF-8".to_string(),
+                "Not valid UTF-8".to_string(),
+                None,
+                None,
+                vec![],
+            ))?;
+        // This function below was taken directly from `uu_cp` to reuse
+        // some of their logic when it came to parsing path arguments,
+        // even tho we already did some of it in the `src, target` for loop above
+        // Nice to combine efforts and make it nicer.
+        let (sources, target) = parse_path_args(paths, &options)?;
+        if let Err(error) = uu_cp::copy(&sources, &target, &options) {
+            match error {
+                // Error::NotAllFilesCopied is non-fatal, but the error
+                // code should still be EXIT_ERR as does GNU cp
+                uu_cp::Error::NotAllFilesCopied => {}
+                // Else we caught a fatal bubbled-up error, log it to stderr
+                // ShellError
+                // _ => uucore::macros::show_error!("{}", error),
+                _ => {
+                    return Err(ShellError::GenericError(
```

Comment:
We could look into integrating `UResult/UError` with the nu system for future utils to make this easier.